code/_helpers/icons.dm

```diff
@@ -744,7 +744,13 @@ as a single icon. Useful for when you want to manipulate an icon via the above a
 			curblend = BLEND_OVERLAY
 			add = icon(I:icon, I:icon_state, I:dir)
 		else // 'I' is an appearance object.
-			add = getFlatIcon(new/image(I), curdir, curicon, curstate, curblend, TRUE)
+			if(istype(A,/obj/machinery/atmospherics) && I in A.underlays)

```

Comment:
This could probably be shoved to follow the else directly. There's a minor performance gain to be had from if/elif/elif/else over if/else(if/elif/else).